task/catlin-validate/0.1/catlin-validate.yaml

```diff
@@ -0,0 +1,112 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: catlin-validate
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/categories: Testing
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: "tekton task yaml validation"
+    tekton.dev/platforms: "linux/amd64"
+    tekton.dev/displayName: catlin
+spec:
+  description: >-
+    Catlin(https://github.com/tektoncd/catlin) is a command-line tool that Lints Tekton Resources and Catalogs.
+    Catlin Validate task supports running catlin validate for
+      1. An isolated task yaml that you want to submit to tekton catalog
+      2. A set of tasks that you want to submit to tekton catalog
+  params:
+    - name: outputFile
+      description: File that contains final catlin output. Catlin output will be stored in /workspaces/catlin-output/$(params.outputFile)
+      type: string
+      default: catlin-output.txt
+    - name: inputFile
+      description: |
+        File that contains catlin input. This file contains relative paths to the task yaml's delimited by newline
+        this file will be read from /workspaces/catlin-input/$(params.inputFile) and catlin validate will be applied to each file
+        example if file contains below paths (as you can see the paths are separated by newlines)
+        task/git-clone/0.9/git-clone.yaml
+        task/git-clone/0.1/git-comment.yaml
+        it will run catlin on the above paths in the source workspace one after another and store its output in catlin-output workspace
+        final command run will be 
+        'catlin validate $(workspaces.source.path)/task/git-clone/0.9/git-clone.yaml >> $(workspaces.catlin-output.path)/output.txt'
+        'catlin validate $(workspaces.source.path)/task/git-clone/0.1/git-comment.yaml >> $(workspaces.catlin-output.path)/output.txt'
```

Comment:
Workspaces I feel are useful for task to task communication. I Prefer workspaces over results for large inputs when one tasks needs to pass data to another task. As catlin-validate task can accept multiple inputs, the input paths can be long. 
Imagine another task that generates the paths on which we need to run catlin-validate
If I use array params, then this other task needs to use results to pass on the input array using. Since inputs can be long they can exceed kubernetes termination string
Instead If we use a volume mount and a file we do not have any such restrictions, which makes the task work for large inputs
Let me know what you think

Reply:
well, due to `affinity-assistance`, we won't be able to mount more than 1 pvc at a time (@vdemeester please correct me if I am wrong 😅), if the former is true then before starting the pipeline, we'll have to mount a ConfigMap containing the input file. I guess this approach can restrict the users who would like to first find the tasks they want to validate using catlin and then pass the list to this task? One solution to this can be binding one pvc workspace with both `inputFile` and `source` workspace and then go ahead but if this is the idea then can't we just have one workspace `source` in the task where we check for the input file ? This goes for `catlin-output` workspace as well. 
@vdemeester @piyush-garg please let me know if there is some gap from my side 😅

Reply:
@vinamra28 The PVC for source and for catlin-input ideally should be different PVC's as they are for completely different purpose. I have created tasks where I have bounded multiple PVC's into different workspaces for a task.

Irrespective I think both these workspaces are readOnly so even mounting the same PVC for these workspaces should work, it is completely upto the user. For this task I think I should mark these workspaces as readOnly as nothing is written to these workspaces and they are merely inputs to catlin-validate task

> we won't be able to mount more than 1 pvc at a time

Why would an affinity assistance restrict mounting more than 1 PVC at a time? 

For instance an example pipeline would be

Pipeline ---> Task1 -> Task2 -> catlin-validate
Task1 -> clones takton catalog in PVC1 workspace
   |
   |
Task2 -> Gets all the tasks that need to run catlin-validate and creates input.txt file in PVC2 
  |
  |
Task3 (catlin-validate)
  -> Loads source from PVC1
  -> Loads catlin-input from PVC2
  
For above since tasks share same workspace backed by PVC the affinity assistant will probably schedule the pods executing the task on the same Kubernetes Node, but it should not restrict  creating more than 1 PVC at a time. Let me know if my understanding is correct. I am also happy to write a test case to validate this

The reason for separating workspaces out is isolation of concern. 

Source workspace refers to the source code which contains all the tekton-catalog tasks (for instance source can contain the git clone of [tekton catalog](https://github.com/tektoncd/catalog) or your internal company tekton catalog)

catlin-input workspace contains the input of paths of task files on which we want to run catlin-validate.

Both these workspaces can be backed with PVC's.

Reply:
@vinamra28 any suggestions comments on above?